File: fixtures/scale/php/large/src/Families/Rot13/Rot13Descriptor.php
<?php

namespace CipherLab\Families\Rot13;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together Rot13Cipher and
// Rot13Vectors under the "classical" suite.
class Rot13Descriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'rot13';
    }

    public function suite(): string
    {
        return 'classical';
    }

    public function cipher(): Cipher
    {
        return new Rot13Cipher();
    }

    public function vectors(): array
    {
        return Rot13Vectors::all();
    }
}

File: fixtures/scale/objc/large/Families/Rot13/Rot13Descriptor.m
#import "Rot13Descriptor.h"
#import "Rot13Cipher.h"
#import "Rot13Vectors.h"

@implementation Rot13Descriptor

- (NSString *)family {
    return @"rot13";
}

- (NSString *)suite {
    return @"classical";
}

- (id<CLCipher>)cipher {
    return [[Rot13Cipher alloc] init];
}

- (NSArray *)vectors {
    return [Rot13Vectors all];
}

@end
